src/monodroid/jni/xamarin_getifaddrs.cc

```diff
@@ -279,6 +285,15 @@ _monodroid_getifaddrs_init ()
 int
 _monodroid_getifaddrs (struct _monodroid_ifaddrs **ifap)
 {
+	if (!initialized) {
+		init_lock.lock ();
```

Comment:
I'm shocked/surprised that this has explicit `lock()`/`unlock()` methods and not a constructor/destructor so that the lock is released in case of exceptions...

Reply:
We don't throw exceptions and we compile without exceptions enabled, so it's not a concern.  But to make you feel better about it, I'll add a lock guard :)